### DISCUSSION TITLE
Multiple TinyMCE instances in Publii's WYSIWYG editor screen?

### DISCUSSION THREAD

**Discussion by grinklelot** [GetPublii/Publii#1715] [Help] [answered]
I'm trying to add some tabbed pages to my site, but have the editor set up so that it's not hard for people without coding knowledge to enter content.

Is there some way to have multiple WYSIWYG boxes? Basically, two separate MCE instances -- where one is for Tab 1 and the other is for Tab 2. Then I could call these via variables in my template.

I found the TinyMCE docs for this, but wasn't sure if it could be adapted. https://www.tiny.cloud/docs/tinymce/latest/multiple-editors/

**Comment by candidexmedia** [comment, marked as answer]
Do you mean tabs like this?

![tab-examples](https://github.com/user-attachments/assets/5e66c8aa-466a-492a-8e88-47dc8beefe90)

Unfortunately, there's no "graphical" way of displaying more complex layouts like that in a meaningful way in the WYSIWYG editor (AFAIK).

That said, I'd suggest the following:

Find an HTML + CSS code snippet for the tabbed groups that works for you, on [Codepen](https://codepen.io/markcaron/pen/MvGRYV) or [Free Frontend](https://freefrontend.com/css-tabs/) or the like.

Then, find a way for your user to easily insert it. The [Template Plugin](https://gist.github.com/candidexmedia/367711a80274b4c005fd5983df97d20c#file-2-adding-a-template-plugin-for-reusable-chunks-md) could be something to look into. The start and end of `divs` for each tab might be hard to distinguish in the editor, though, so consider creating some [Post Editor CSS rules](https://getpublii.com/dev/post-editor-api/#customposteditorcssstyles) (they'll only appear in the editor).

Maybe @bjazmoore's plugin could help as well: https://github.com/bjazmoore/Key-Replacement. Your users would just have to enter the special "keyword" to designate where tab content starts and ends, and you could even combine this with the Template Plugin.

If the presence of tabs is consistent across many pages, and the number of tabs and position is the same (i.e. at the top of the page, below the title), you could consider hard-coding the tabbed group code in the post.hbs (in the theme files), and making the content editable via the [Theme Settings API > Post Settings](https://getpublii.com/dev/theme-settings-api/#howtocreatepostsettings).

Last note: make sure [tabbed groups are the right call, UX-wise](https://www.nngroup.com/articles/tabs-used-right/).

**Reply by grinklelot** [reply]
Thank you so much.

Yes, that is the style of tabs I'm using. I will try some of these, I think they could all potentially work very well in combination.

For your last suggestion (editing post.hbs) -- it seems like I'd be able to call a WYSIWG field this way? The tabs are consistent across many pages, so I'll try that out.

Thanks again!

**Reply by candidexmedia** [reply]
@grinklelot I double-checked the docs, and WYSIWYG text fields are unfortunately only available for theme settings. For post settings, you get the following options:

![image](https://github.com/user-attachments/assets/7c87c8d0-5ca0-460e-b68f-b84f8943ce86)

The [text area field](https://getpublii.com/dev/textarea/) accepts plain text, although HTML `<tags>` can be added if you add `{{{ three sets of curly brackets }}}` in your `post.hbs`.

So... a template plugin might be a better call if your user will need to add styling to the text?

I'd love to know what you settle on, and how it looks once you've implemented it :)

**Reply by grinklelot** [reply]
Thank you. I'll be trying a few things, you never know how "easy" they are to use until you get them in front of you haha.

The post settings thing seems like it'd be easiest... I kind of wish I could move the field underneath the main editor instead of leaving it on the side. But I'll live.

Thank you again! I'll follow up when I figure it out.
